File: 214/214.py
class Solution:
    def shortestPalindrome(self, s: str) -> str:
        #from the editorial 
        n = len(s)
        rs = s[::-1]
        for i in range(n):
            if s[:n-i] == rs[i:]:
                return rs[:i] + s
        return ""

File: 214/214.java
class Solution {
    public String shortestPalindrome(String s) {
        int n = s.length();
        String rs = new StringBuilder(s).reverse().toString();
        for(int i = 0; i < n; i++){
            if (s.substring(0,(n-i)).equals(rs.substring(i))){
                return rs.substring(0,i) + s;
            }
        }
        return "";
    }
}
